File: client-scalaz/src/main/scala/org/openapitools/client/api/VehicleType.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import VehicleType._

case class VehicleType (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
business: Option[Retailer],
hub: Option[ServiceHub],
name: Option[String],
width: Option[Double],
height: Option[Double],
depth: Option[Double],
volume: Option[Double],
maxWeight: Option[Double],
vehicleCargoSettings: Option[List[VehicleCargoSetting]])

object VehicleType {
  import DateTimeCodecs._

  implicit val VehicleTypeCodecJson: CodecJson[VehicleType] = CodecJson.derive[VehicleType]
  implicit val VehicleTypeDecoder: EntityDecoder[VehicleType] = jsonOf[VehicleType]
  implicit val VehicleTypeEncoder: EntityEncoder[VehicleType] = jsonEncoderOf[VehicleType]
}

File: client-clojure/src/sirqul_io_t_platform/specs/vehicle_type.clj
(ns sirqul-io-t-platform.specs.vehicle-type
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.retailer :refer :all]
            [sirqul-io-t-platform.specs.service-hub :refer :all]
            [sirqul-io-t-platform.specs.vehicle-cargo-setting :refer :all]
            )
  (:import (java.io File)))


(def vehicle-type-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :business) retailer-spec
   (ds/opt :hub) service-hub-spec
   (ds/opt :name) string?
   (ds/opt :width) float?
   (ds/opt :height) float?
   (ds/opt :depth) float?
   (ds/opt :volume) float?
   (ds/opt :maxWeight) float?
   (ds/opt :vehicleCargoSettings) (s/coll-of vehicle-cargo-setting-spec)
   })

(def vehicle-type-spec
  (ds/spec
    {:name ::vehicle-type
     :spec vehicle-type-data}))
